### DISCUSSION TITLE
How to execute `.Rprofile` of each document before rendering whole project?

### DISCUSSION THREAD

**Discussion by JackCaster** [quarto-dev/quarto-cli#10681] [Q&A]
I have a collection of blog posts, each with a specific renv lockfile. The only way I can make quarto load the correct renv specification before rendering each posts is to include an R code chunk in the `.qmd` file for the post with

```qmd
#| file: '.Rprofile'
```

or 
```
source('.Rprofile')
```

or 
```
{{< include .Rprofile >}}
```

Is there an alternative way to make quarto load the `.Rprofile` automatically?

This is related to https://github.com/quarto-dev/quarto-cli/discussions/6864, https://github.com/rstudio/blogdown/issues/469 but I was not satisfied with the suggestions there, unfortunately

**Comment by mcanouil** [comment]
Quarto does not support document specific environments.
The answer remains exactly the same as in the posts you shared.

This is a duplicate of:

- https://github.com/quarto-dev/quarto-cli/discussions/6864

**Comment by JackCaster** [comment]
Ok I see. But `.Rprofile` can be used for other R-specific settings/options that are usually loaded by default when R is started (let's forget renv for a moment). Is `.Rprofile` loaded automatically by Quarto before each rendering? I can't understand if it does...

**Reply by mcanouil** [reply]
yes, if at the root of the project.
When and where R starts if there is an `.Rprofile`, it is loaded, but that's R behaviour unrelated to Quarto.

**Reply by AlbertoFabbri93** [reply]
Is this the case? It doesn't seem to load `.Rprofile` if is not in the same folder as the Quarto file.

**Reply by mcanouil** [reply]
I can guarantee it is, I'm using `.Rprofile` in all my Quarto projects.
The `.Rprofile` needs to be located at the execution directory which is defined either by the document or `_quarto.yml`.

https://quarto.org/docs/projects/code-execution.html#working-dir
